onnx/shape_inference/implementation.cc

```diff
@@ -191,15 +191,13 @@ static void InferShapesImpl(
     // Consider the tensors from the initializer
     TypeProto *initializerType = new TypeProto();
     TypeProto_Tensor* initializerTensorType = initializerType->mutable_tensor_type();
+    initializerTensorType->set_elem_type(tp.data_type());
     // set the shape according to the initializer shape info
-    // if the shape info of initializer is not empty
-    if (tp.dims_size() != 0) {
-      initializerTensorType->set_elem_type(tp.data_type());
-      TensorShapeProto* shape = initializerTensorType->mutable_shape();
-      for (int i = 0 ; i < tp.dims_size(); ++i) {
-        shape->add_dim()->set_dim_value(tp.dims(i));
-      }
+    TensorShapeProto* shape = initializerTensorType->mutable_shape();
```

Comment:
I think this check -> tp.dims_size() !=0 is needed before output tensor shape is initialized?
This can cause issues because has_shape for this tensor will return true from here onwards and depending what the calling code (whether it does necessary checks or not ) it can cause segfaults

Reply:
The dims_size check here is not related to whether the shape is known or not known. Here, the shape is known exactly. If dims_size is 0, it means it is rank 0 (a scalar). This check is different from a "has_shape" kind of check.

Reply:
I added `tp.dims_size() != 0` because 3 tests in `shape_inference_test` will fail without this. My thought was same as @askhade that this can check whether the shape is initialized.
If `dims_size=0` means a scalar, there might be other potential bugs in `splitToSequence` because the ranks of initializer and input are different: `RuntimeError: Inferred shape and existing shape differ in rank: (0) vs (1)
`.

Reply:
It is possible that there are bugs elsewhere (in requiring rank 0 or rank 1 for supposedly-scalar inputs). Or, if the spec calls for rank 1, and the model uses rank 0, it is an error in the model. Adding the check here will only mask problems elsewhere, I think.

Reply:
The split_to_sequence testcases had an issue. I fixed them. Hope this goes through the CI.